### ISSUE TITLE
 can't copy 'etc/bash_completion.d/pyang': doesn't exist or not a regular file

### ISSUE THREAD

**Issue by adudek16** [mbj4668/pyang#174]
I am running into this error when running python setup.py install.
I see that the directory is empty. I am not sure if I am doing something wrong

grove:/tmp/pyang-1.6/build/bdist.linux-x86_64/egg/etc$ ls -la bash_completion.d/
total 8
drwxr-xr-x 2 root root 4096 Nov  4 11:18 .
drwxr-xr-x 3 root root 4096 Nov  4 11:18 ..
grove:/tmp/pyang-1.6/build/bdist.linux-x86_64/egg/etc$

**Comment by llhotka** [comment]
Please try to install it via pip, e.g
```
$ sudo pip install pyang
```
Do you see the same problem?

**Comment by adudek16** [comment]
No. I do not.

**Comment by bob-smith** [comment]
I have the same issue when installing 1.6. This is failing on Ubuntu 12.04

from /root/.pip/pip.log

```
/usr/bin/pip run on Tue Dec  1 09:06:51 2015
Downloading/unpacking pyang==1.6
  Getting page http://pypi.python.org/simple/pyang
  URLs to search for versions for pyang==1.6:
  * http://pypi.python.org/simple/pyang/1.6
  * http://pypi.python.org/simple/pyang/
  Getting page http://pypi.python.org/simple/pyang/1.6
  Getting page http://pypi.python.org/simple/pyang/
  Could not fetch URL http://pypi.python.org/simple/pyang/1.6: HTTP Error 404: Not Found
  Will skip URL http://pypi.python.org/simple/pyang/1.6 when looking for download links for pyang==1.6
  Analyzing links from page https://pypi.python.org/simple/pyang/
    Skipping link https://pypi.python.org/packages/py2.py3/p/pyang/pyang-1.6-py2.py3-none-any.whl#md5=db132b7f0d0a4c146f14b6b8478bbb86 (from https://pypi.python.org/simple/pyang/); unknown archive format: .whl
    Found link https://pypi.python.org/packages/source/p/pyang/pyang-1.6.tar.gz#md5=503df793cf91de6d3ee48b5508007848 (from https://pypi.python.org/simple/pyang/), version: 1.6
  Downloading from URL https://pypi.python.org/packages/source/p/pyang/pyang-1.6.tar.gz#md5=503df793cf91de6d3ee48b5508007848 (from https://pypi.python.org/simple/pyang/)
  Running setup.py egg_info for package pyang


SNIP

    byte-compiling /usr/local/lib/python2.7/dist-packages/pyang/yin_parser.py to yin_parser.pyc
    byte-compiling /usr/local/lib/python2.7/dist-packages/pyang/statements.py to statements.pyc
    running install_data
    error: can't copy 'etc/bash_completion.d/pyang': doesn't exist or not a regular file
    Complete output from command /usr/bin/python -c "import setuptools;__file__='/tmp/build/pyang/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --single-version-externally-managed --record /tmp/pip-pOwh01-record/install-record.txt:
    running install

running build

SNIP

byte-compiling /usr/local/lib/python2.7/dist-packages/pyang/yin_parser.py to yin_parser.pyc

byte-compiling /usr/local/lib/python2.7/dist-packages/pyang/statements.py to statements.pyc

running install_data

error: can't copy 'etc/bash_completion.d/pyang': doesn't exist or not a regular file

----------------------------------------
Command /usr/bin/python -c "import setuptools;__file__='/tmp/build/pyang/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --single-version-externally-managed --record /tmp/pip-pOwh01-record/install-record.txt failed with error code 1
Exception information:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 126, in main
    self.run(options, args)
  File "/usr/lib/python2.7/dist-packages/pip/commands/install.py", line 228, in run
    requirement_set.install(install_options, global_options)
  File "/usr/lib/python2.7/dist-packages/pip/req.py", line 1093, in install
    requirement.install(install_options, global_options)
  File "/usr/lib/python2.7/dist-packages/pip/req.py", line 566, in install
    cwd=self.source_dir, filter_stdout=self._filter_install, show_stdout=False)
  File "/usr/lib/python2.7/dist-packages/pip/__init__.py", line 255, in call_subprocess
    % (command_desc, proc.returncode))
InstallationError: Command /usr/bin/python -c "import setuptools;__file__='/tmp/build/pyang/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --single-version-externally-managed --record /tmp/pip-pOwh01-record/install-record.txt failed with error code 1

```

**Comment by miott** [comment]
I have this issue trying to pip install on my macbook pro in a virtualenvwrapper.  This is what I discovered:

There seems to be a problem with in pyang setup.py.  I added the correct path to pyang in the pyang setup.py data_files and it worked.

Changed From:
      data_files=[
            ('share/man/man1', man1),
            ('share/yang/modules/iana', modules_iana),
            ('share/yang/modules/ietf', modules_ietf),
            ('share/yang/xslt', xslt),
            ('share/yang/images', images),
            ('share/yang/schema', schema),
            ('etc/bash_completion.d', ['etc/bash_completion.d/pyang']),
            ]
      )
To:
      data_files=[
            ('share/man/man1', man1),
            ('share/yang/modules/iana', modules_iana),
            ('share/yang/modules/ietf', modules_ietf),
            ('share/yang/xslt', xslt),
            ('share/yang/images', images),
            ('share/yang/schema', schema),
            #('etc/bash_completion.d', ['etc/bash_completion.d/pyang']),
            ('etc/bash_completion.d', ['bin/pyang']),
            ]
      )
~       

When you specify a relative path in data_files, it is suppose to extend sys.prefix.  As you can see from the install log, it actually runs setup from <virtualenv dir>/build/pyang/setup.py

    Complete output from command /Users/miott/ws/ydk/bin/python -c "import setuptools;__file__='/Users/miott/ws/ydk/build/pyang/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/42/cpckzs1d6jg63l_b2pgvwn300000gn/T/pip-ms2eS6-record/install-record.txt --single-version-externally-managed --install-headers /Users/miott/ws/ydk/bin/../include/site/python2.7:
    running install

The current setup.py data_files specification says “copy sys.prefix+/etc/bash_completion.d/pyang to sys.prefix+/etc/bash_completion.d/“.  By changing it to 'bin/pyang', the command changes to "copy sys.prefix+/etc/bash_completion.d/pyang to sys.prefix+/bin/pyang“.

Have not tried this on Ubuntu so don't know if it would fix or break that install.

**Comment by miott** [comment]
Update:
Using virtualenvwrapper on my macbook pro, when I mkvirtualenv, I noticed pip --version in the new virtualenv is 1.2.1.  I have the pyang pip install issue using this version of pip.  When I deactivate out of a virtualenv, the pip version in my system python library is 7.1.0 and the pyang pip install works fine.  Just have to figure out where my virtualenvs are picking up the old pip version.

Does not explain what I found in the setup.py in data_files.  A different version of pip should not have fixed that issue.  I think the new version of pip is intelligent enough to work around that problem.

**Comment by jefftant** [comment]
On El Capitan
Replacing:
etc/bash_completion.d/pyang by
bin/pyang
fixes the issue

**Comment by ganesh-karthick** [comment]
Running pip install --upgrade pip within virtualenv fixed this issue.
pip 8.1.2 where it worked

**Comment by llhotka** [comment]
Solved by merging #268.
